### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by rachmatowicz**
This PR included fixes for:
- the bean-management attrobute XML persisting issue
- the persistence context tests in wildfly-ts-integ-basic

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java [hunk 1]
```diff
@@ -145,4 +107,57 @@ public void testNonExtended() throws Exception {
...
+            operation.get(OP_ADDR).set(address);
+            operation.get("name").set("max-size");
+            operation.get("value").set(10000);
+
+            ModelNode result = managementClient.getControllerClient().execute(operation);
+            Assert.assertEquals(SUCCESS, result.get(OUTCOME).asString());
+            ServerReload.reloadIfRequired(managementClient);
+        }
+
+        private static ModelNode getAddress() {
```
Comment (pferraro) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:154]: Why not utilize PathAddress?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java [hunk 2]
```diff
@@ -145,4 +107,57 @@ public void testNonExtended() throws Exception {
...
+    static class StatefulUnitTestCaseSetup implements ServerSetupTask {
+
+        /*
+         * Set the max-size attribute of the bean-management provider to 1 to force passivation.
+         */
+        @Override
+        public void setup(final ManagementClient managementClient, final String containerId) throws Exception {
+            ModelNode address = getAddress();
+
+            ModelNode operation = new ModelNode();
```
Comment (pferraro) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:125]: Use Util.createWriteAttributeOperation(...)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java [hunk 3]
```diff
@@ -145,4 +107,57 @@ public void testNonExtended() throws Exception {
...
+        }
+
+        /*
+         * Return max-size to its configured value (10,000).
+         */
+        @Override
+        public void tearDown(final ManagementClient managementClient, final String containerId) throws Exception {
+            ModelNode address = getAddress();
+
+            ModelNode operation = new ModelNode();
```
Comment (pferraro) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:143]: Use Util.createWriteAttributeOperation(...)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java [hunk 4]
```diff
@@ -145,4 +107,57 @@ public void testNonExtended() throws Exception {
...
+            operation.get(OP).set("write-attribute");
+            operation.get(OP_ADDR).set(address);
+            operation.get("name").set("max-size");
+            operation.get("value").set(10000);
```
Comment (pferraro) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:147]: I would include a comment to make others aware that this is a configuration default - thus any changes to the default configuration should be reflected here.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemXMLPersister.java [hunk 5]
```diff
@@ -569,7 +569,8 @@ private void writeDistributableCaches(XMLExtendedStreamWriter writer, ModelNode
                 writer.writeStartElement(EJB3SubsystemXMLElement.DISTRIBUTABLE_CACHE.getLocalName());
                 ModelNode distributableCache = property.getValue();
                 writer.writeAttribute(EJB3SubsystemXMLAttribute.NAME.getLocalName(), property.getName());
-                DistributableCacheFactoryResourceDefinition.Attribute.BEAN_MANAGEMENT.getDefinition().marshallAsElement(distributableCache, writer);
+                AttributeDefinition definition = DistributableCacheFactoryResourceDefinition.Attribute.BEAN_MANAGEMENT.getDefinition();
+                definition.getMarshaller().marshallAsAttribute(definition, distributableCache, false, writer);
```
Comment (pferraro) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemXMLPersister.java:573]: Can you future-proof this?
i.e.
```java
for (Attribute attribute : EnumSet.allOf(DistributableCacheFactoryResourceDefinition.Attribute.class)) {
    attribute.getDefinition().getMarshaller().marshallAsAttribute(attribute.getDefinition(), distributableCache, false, writer);
}
```

**Review by pferraro — CHANGES_REQUESTED** [review]

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java (hunk 1 above)
Comment (rachmatowicz) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:154]: Done

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java (hunk 2 above)
Comment (rachmatowicz) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:125]: Done

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java (hunk 3 above)
Comment (rachmatowicz) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:143]: Done.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java (hunk 4 above)
Comment (rachmatowicz) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/persistencecontext/StatefulUnitTestCase.java:147]: Done.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemXMLPersister.java (hunk 5 above)
Comment (rachmatowicz) [reply ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemXMLPersister.java:573]: Done.

**Pull request closed without merging**